### ISSUE TITLE
new style rule support

### ISSUE THREAD

**Issue by michael-conway** [DICE-UNC/jargon#379]
rule call style with 3 json structures (supports gallery) per this example

```json
*params:

{
  "logical_path": string,
  "offset": integer,
  "length": integer
}
*config:

{
}
*out:

{
  "logical_path": string,
  "items": [
    {
      "id": integer,
      "name": string,
      "last_modified": string,
      "thumbnail": string
    }
  ]
}
```

**Comment by michael-conway** [comment]
There are some difficulties here with how the actual rules protocol works, and how to manage parameter overrides in the JSON. This will be something to smooth out with @trel and in the meantime we punted in the gallery rule for now. This might go into the next feature release.

**Comment by jasoncoposky** [comment]
@michael-conway We are just passing the json as a string representations, not as a specific type.  The signature is simply a rule name with three parameters which are strings, e.g.

```
irods_policy_do_the_thing(*parameters, *configuration, *out_variable)
{
   *out = "{ \"key\" : \"value\",
                  \"foo\" : \"bar\"
                }
}
```
Is there some Jargon issue getting in the way?

**Comment by michael-conway** [comment]
Yes, it got too messy w json and STR_PI I can show you are some point but
can be addressed but was getting in the way in the near term. Lots of
escaping, not a great story on allowing a rule w an INPUT line and a
desired param override of a variable in the input line. Quotes in json
making rule engine Very.Mad.
